### PR TITLE
Make InMemoryViewStore thread safe

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/view/InMemoryViewStore.java [hunk 1]
```diff
@@ -32,26 +35,46 @@
...
   @Override
   public boolean exists(Id.Stream.View viewId) {
-    return views.contains(viewId, viewId.getStream());
+    Lock lock = viewsLock.readLock();
+    lock.lock();
+    try {
+      return views.contains(viewId, viewId.getStream());
+    } finally {
+      lock.unlock();
+      lock.unlock();
```
Comment (jwang47) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/view/InMemoryViewStore.java]: Why 2 unlock calls?

**Comment by jwang47** [comment]
One minor comment otherwise LGTM

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/view/InMemoryViewStore.java (hunk 1 above)
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/view/InMemoryViewStore.java]: hum, no idea. Maybe copy-paste issue.

Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/view/InMemoryViewStore.java]: fixed.

**Comment by bdmogal** [comment]
Build passed - https://builds.cask.co/browse/CDAP-DUT2856/latest. Merged so we can plan for release branch cutting better.

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT2856-2

**Pull request merged**